src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -278,8 +278,12 @@ export class NotebookProviderInfoStore extends Disposable {
 			throw new Error(`notebook type '${info.id}' ALREADY EXISTS`);
 		}
 		this._contributedEditors.set(info.id, info);
-		const editorRegistration = this._registerContributionPoint(info);
-		this._contributedEditorDisposables.add(editorRegistration);
+		let editorRegistration: IDisposable | undefined;
+		// Don't overwrite editor contributions if they come from elsewhere
+		if (!info.externalEditor) {
```

Comment:
I wonder if we can have better name to indicate that we only want to register the notebook provider info (for resolving notebook model) but we don't want to register an editor.